crates/valence/src/instance/chunk.rs

```diff
@@ -46,6 +53,97 @@ struct Section {
     section_updates: Vec<VarLong>,
 }
 
+/// Represents a block with an optional block entity
+#[derive(Clone, Debug)]
+pub struct Block {
+    state: BlockState,
+    /// Nbt of the block entity
+    nbt: Option<Compound>,
+}
+
+impl Block {
+    pub const AIR: Self = Self::new(BlockState::AIR);
+
+    pub const fn new(state: BlockState) -> Self {
+        Self {
+            state,
+            nbt: match state.block_entity_kind() {
+                Some(_) => Some(Compound::new()),
+                None => None,
+            },
+        }
+    }
+
+    /// # Panics
+    /// This function panics if the specified [`BlockState`] does not support a
+    /// block entity yet `nbt` is [`Option::Some`]
+    pub const fn with_nbt(state: BlockState, nbt: Option<Compound>) -> Self {
+        if state.block_entity_kind().is_none() {
+            assert!(nbt.is_none(), "blockstate does not support nbt");
```

Comment:
Just doing that silently isn't that good of an idea I feel like

Reply:
I would normally agree with you in situations like these, but considering that we have no way to verify the actual contents of the compound are correct, panicking feels inconsistent. In other words, we're already silently ignoring errors by accepting nonsensical compounds, so setting `nbt` to `None` silently is consistent with this behavior.

In any case, if we must fail loudly it should probably return `Option<Self>` instead of panicking. But I would prefer to just set `nbt` to `None`.

Side note: Should this function accept a `Compound` instead of an `Option<Compound>`? Passing in `None` is equivalent to calling `new` so you might as well call `new` instead.